src/components/modals/authentification/ModalAuthConnexionView.tsx

```diff
@@ -0,0 +1,101 @@
+/**
+ * Modal avec formulaire de connexion quand user deja inscrit
+ */
+
+import { firebaseSignInUser } from '@/api/authentification';
+import { SubmitHandler, useForm } from 'react-hook-form';
+
+type Props = {
+  setContenuModal: React.Dispatch<
+    React.SetStateAction<
+      'init' | 'connexion' | 'inscription' | 'password-forget'
+    >
+  >;
+};
+
+type LoginFormTypeConnexion = {
+  email: 'string';
+  password: 'string';
+};
+
+const ModalAuthConnexionView = ({ setContenuModal }: Props) => {
+  const { handleSubmit, register, setError, reset } =
+    useForm<LoginFormTypeConnexion>();
+
+  /**
+   * 2.
+   */
+  const handleSignInUser = async ({ email, password }: LoginFormTypeConnexion) => {
+    const { error } = await firebaseSignInUser(email, password);
+    if (error) {
+      console.log("Error handleSignInUser")
+      //toast.error(error.message);
+      return;
+    }
+    reset();
+    setContenuModal('init')
+  };
+
+  /**
+   * 1. Test longueur du password
+   * Envoi des data
+   */
+  const onSubmit: SubmitHandler<LoginFormTypeConnexion> = async (data) => {
+    const { password } = data;
+    if (password.length < 7) {
+      setError('password', {
+        type: 'manuel',
+        message: 'Le mot de passe doit comporter 7 caractères minimum',
+      });
+      return;
+    }
+    handleSignInUser(data);
+  };
+
+  return (
+    <>
```

Comment:
Message de Sonarlint : A fragment with only one child is redundant.sonarlint(typescript:S6749)

Reply:
corrigé